xcm/xcm-emulator/src/lib.rs

```diff
@@ -622,14 +626,16 @@ macro_rules! decl_test_parachains {
 						let block_number = <Self as Chain>::System::block_number();
 						let mut relay_block_number = <Self as NetworkComponent>::Network::relay_block_number();
 
-						let _ = <Self as Parachain>::ParachainSystem::set_validation_data(
```

Comment:
Calling `set_validation_data` here is definitely illegal so had to be removed. Tests possibly depended on it, but it lead to an impossible sequence of calls.

Reply:
elaborating a bit more. `init()` is called within `execute_block` the first time it's run.

`set_validation_data` needs to be followed by a `System::finalize` call.
It might be more correct as 
1. initialize
2. set_validation_data
3. finalize
4. initialize

but that is basically just running a full block, which shouldn't be necessary.